runtime/crab/src/migration.rs

```diff
@@ -41,28 +41,32 @@ impl frame_support::traits::OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
 }
 
 fn migrate() -> frame_support::weights::Weight {
-	// core
-	use core::str::FromStr;
-	// substrate
-	use frame_support::pallet_prelude::StorageVersion;
+	// polkadot-sdk
+	use frame_support::traits::{LockableCurrency, ReservableCurrency};
 
-	let _ = migration::clear_storage_prefix(b"PhragmenElection", b"Members", &[], None, None);
-	let _ = migration::clear_storage_prefix(b"PhragmenElection", b"RunnersUp", &[], None, None);
-	let _ = migration::clear_storage_prefix(b"PhragmenElection", b"Candidates", &[], None, None);
-	let _ =
-		migration::clear_storage_prefix(b"PhragmenElection", b"ElectionRounds", &[], None, None);
-	let _ = migration::clear_storage_prefix(b"PhragmenElection", b"Voting", &[], None, None);
-	let _ = migration::clear_storage_prefix(b"TechnicalMembership", b"Members", &[], None, None);
-	let _ = migration::clear_storage_prefix(b"TechnicalMembership", b"Prime", &[], None, None);
+	[
```

Comment:
Can you provides more information about the underlying logic behind this migration? I am not familar with the PhragmenElection module, and I have the following questions:

1. Why reserve then remove_lock, are both of them necessary?
2. `Voter { votes: "[0x203e49868a6ce5c9cebc8ee260ec9840335d7d27]", stake: 99793500000000000000, deposit: 100009600000000000000 }` We don't need to care about the stake field?

I don't have enough knowledge to review this now, undless takes some time to figure out what it does in the PhragmenElection and staking electtion.

Reply:
You just answered your question.

Remove lock for stake. Unreserve for deposit.